packages/react-components/lib/tasks/create-task.tsx

```diff
@@ -976,159 +164,47 @@ function FavoriteTask({
-          },
-        },
-        on_cancel: [],
-      },
-      {
-        activity: {
-          category: 'sequence',
-          description: {
-            activities: [
-              {
-                category: 'perform_action',
-                description: {
-                  unix_millis_action_duration_estimate: 60000,
-                  category: 'delivery_dropoff',
-                  description: {},
-                },
-              },
-            ],
-          },
-        },
-      },
-    ],
-  };
-}
-
-export function defaultDeliveryCustomTaskDescription(
-  taskCategory: string,
-): DeliveryCustomTaskDescription {
-  return {
-    category: taskCategory,
-    phases: [
-      {
-        activity: {
-          category: 'sequence',
-          description: {
-            activities: [
-              {
-                category: 'go_to_place',
-                description: '',
-              },
-              {
-                category: 'perform_action',
-                description: {
-                  unix_millis_action_duration_estimate: 60000,
-                  category: taskCategory,
-                  description: {
-                    cart_id: '',
-                    pickup_zone: '',
-                  },
-                },
-              },
-            ],
-          },
-        },
-      },
-      {
-        activity: {
-          category: 'sequence',
-          description: {
-            activities: [
-              {
-                category: 'go_to_place',
-                description: '',
-              },
-            ],
-          },
-        },
-        on_cancel: [],
-      },
-      {
-        activity: {
-          category: 'sequence',
-          description: {
-            activities: [
-              {
-                category: 'perform_action',
-                description: {
-                  unix_millis_action_duration_estimate: 60000,
-                  category: 'delivery_dropoff',
-                  description: {},
-                },
-              },
-            ],
-          },
-        },
-      },
-    ],
-  };
-}
-
-export function defaultPatrolTask(): PatrolTaskDescription {
-  return {
-    places: [],
-    rounds: 1,
-  };
+function defaultTaskDescription(taskName: string): TaskDescription | undefined {
+  switch (taskName) {
+    case 'delivery_pickup':
+      return makeDefaultDeliveryTaskDescription();
+    case 'delivery_sequential_lot_pickup':
+    case 'delivery_area_pickup':
+      return makeDefaultDeliveryCustomTaskDescription(taskName);
+    case 'patrol':
+      return makeDefaultPatrolTask();
+    case 'delivery':
+      return makeDefaultSimpleDeliveryTaskDescription();
+    case 'clean':
+      return makeDefaultCleanTaskDescription();
+    default:
+      return undefined;
+  }
 }
 
-function defaultTaskDescription(taskCategory: string): TaskDescription | undefined {
-  switch (taskCategory) {
+function taskRequestCategory(taskName: string): string | undefined {
```

Comment:
If we use the minimal "task definition", we would no longer need to do this hardcoded mapping.

Reply:
this mapping should be left in, as it is currently the best source of truth side-by-side the implementation, otherwise we will have to perform a lot more additional error checking and validation on the build-time configuration to ensure we are allowing valid tasks.

this mapping will be removed once we shift to the json schema based task definition